components/formats-gpl/src/loci/formats/in/CellH5Reader.java

```diff
@@ -481,6 +495,20 @@ private void parseStructure() throws FormatException {
 
     for (int s=0; s<seriesNames.size(); s++) {
       store.setImageName(seriesNames.get(s), s);
+      
+      String plate_id =  MetadataTools.createLSID("Plate", 0);
+     
+      store.setPlateID(plate_id, 0);
+      store.setPlateName(seriesPlate.get(s), 0);
+      
+      String well_id =  MetadataTools.createLSID("Well", 0);
+      
+      store.setWellID(well_id, 0, 0);
+      store.setWellExternalIdentifier(seriesWell.get(s), 0, 0);
+      
+      String site_id = MetadataTools.createLSID("WellSample", 0);
+      store.setWellSampleID(site_id, 0, 0, 0);
+      store.setWellSampleIndex(NonNegativeInteger.valueOf(seriesSite.get(s)), 0, 0, 0);
```

Comment:
The Well and WellSample population here is potentially problematic.

The row and column values on Well (```store.setWellRow(...)``` and ```store.setWellColumn(...)```) are required to produce valid OME-XML; these should be set after the Well ID.  Only one Well and WellSample are created, even if multiple wells and sites exist - this means that only the last well and site will be stored in the OME-XML.

While not strictly necessary, it's a good idea to also call  ```store.setWellSampleImageRef(...)```, as this ensures that the Image metadata and the Plate/Well metadata are correctly linked together.

An example of populating a full plate is here:

https://github.com/openmicroscopy/bioformats/blob/develop/components/formats-gpl/src/loci/formats/in/ScanrReader.java#L598 through line 650